app/src/main/kotlin/io/intrepid/skotlinton/base/BaseFragment.kt

```diff
@@ -11,18 +14,34 @@ import android.view.ViewGroup
 import butterknife.ButterKnife
 import butterknife.Unbinder
 import io.intrepid.skotlinton.SkotlintonApplication
+import io.reactivex.disposables.CompositeDisposable
 import timber.log.Timber
 
-abstract class BaseFragment<out P : BaseContract.Presenter> : Fragment(), BaseContract.View {
+abstract class BaseFragment<out VM : BaseViewModel> : Fragment() {
 
     protected val skotlintonApplication: SkotlintonApplication
-        get() = activity!!.application as SkotlintonApplication
+        get() = requireActivity().application as SkotlintonApplication
     protected abstract val layoutResourceId: Int
 
-    protected val presenter: P by lazy(LazyThreadSafetyMode.NONE) {
-        val configuration = skotlintonApplication.getPresenterConfiguration()
-        createPresenter(configuration)
+    protected val onPauseDisposable = CompositeDisposable()
+    protected val onStopDisposable = CompositeDisposable()
+    protected val onDestroyViewDisposable = CompositeDisposable()
+    protected val onDestroyDisposable = CompositeDisposable()
+
+    @Suppress("UNCHECKED_CAST")
```

Comment:
Can we move this suppression to line 36 so it's scoped only to where the cast happens instead of the whole `viewModel` declaration?

Reply:
Actually, I guess there's two unchecked casts so top-level suppression does make some sense.